dimod/binary_quadratic_model.py

```diff
@@ -1612,6 +1612,92 @@ def from_json(cls, obj):
 
         return json.load(obj,  object_hook=bqm_decode_hook)
 
```

Comment:
PR metacomment: I see the pattern of adding convenience to/from methods escalating quickly here.

To be fair, this `to_bson` does add a little bit of an optimization effort on top of `BSON.encode(bqm.to_dict())`, it's the need for BSON support in Ocean community I'm not certain about.

@arcondello Maybe it would make sense to consider having to/from convenience methods as some kind of BQM plugins, since BQM implementation is already pretty large, and it'll only grow over time.

Reply:
This is already in a sense a plugin, the methods will fail unless the dependencies are installed. I am not sure I see an issue with the number of these things propagating within reason. Especially, as in this case, when they add actual functionality. The BQM object is afterall designed for convenience and transforming to and from other formats is one of its primary purposes.

Reply:
In a very stretched sense. All methods still live inside the BQM class, inside one large file.

Even if formats conversion is one of the primary purposes of BQM (wasn't aware of that), I'll argue there's value in splitting those methods somehow, yanking them out of base base def. Just imagine the case of hundreds of conversion methods...